skribenten-web/frontend/src/Brevredigering/LetterEditor/actions/toggleBulletList.ts

```diff
@@ -0,0 +1,273 @@
+ *  Det er fordi vi må styre hvor den nye contenten skal havne, samt resten av innholdet i punktlisten.
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOff = (draft: Draft<LetterEditorState>, itemContentIndex: ItemContentIndex) => {
+  const thisBlock = draft.redigertBrev.blocks[itemContentIndex.blockIndex];
+  const thisItemList = thisBlock.content[itemContentIndex.contentIndex] as ItemList;
+
+  if (itemContentIndex.itemIndex === 0) {
+    toggleBulletListOffAtTheStartOfItemList({ draft, itemContentIndex });
+  } else if (itemContentIndex.itemIndex > 0 && itemContentIndex.itemIndex < thisItemList.items.length - 1) {
+    toggleBulletListOffBetweenListElements({ draft, itemContentIndex });
+  } else if (itemContentIndex.itemIndex === thisItemList.items.length - 1) {
+    toggleBulletListOffAtTheEndOfItemList({ draft, itemContentIndex });
+  }
+};
+
+/**
+ * Fjerner det første punktet fra en punktliste, og legger det som en vanlig text-content i blocken.
+ * Legger tilbake resten av punktene i punktlisten
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOffAtTheStartOfItemList = (args: {
+  draft: Draft<LetterEditorState>;
+  itemContentIndex: ItemContentIndex;
+}) => {
+  const thisBlock = args.draft.redigertBrev.blocks[args.itemContentIndex.blockIndex];
+  const thisItemList = thisBlock.content[args.itemContentIndex.contentIndex] as ItemList;
+  const thisItem = thisItemList.items[0];
+
+  thisItemList.items.splice(0, 1);
+  deleteElement(thisItem, thisItemList.items, thisItemList.deletedItems);
+
+  thisBlock.content.splice(
+    args.itemContentIndex.contentIndex,
+    thisItemList.items.length === 0 ? 1 : 0,
+    ...thisItem.content,
+  );
+  deleteElement(thisItemList, thisBlock.content, thisBlock.deletedContent);
+
+  args.draft.focus = {
+    blockIndex: args.itemContentIndex.blockIndex,
+    contentIndex: args.itemContentIndex.contentIndex + (args.itemContentIndex.itemContentIndex ?? 0),
+    cursorPosition: args.draft.focus.cursorPosition,
+  };
+};
+
+/**
+ * Fjerner et punkt fra en punktliste, som ikke er det første eller siste punktet i punktlisten
+ * legger det som en vanlig text-content i blocken.
+ * Bygger 2 nye itemLists, med innhold som er før og etter det punktet som skal fjernes, og setter de inn i blocken
+ *
+ * Fordi vi gjør en såpass stor endring i dokument strukturen, Så må vi oppdatere fokuset til editorstaten til å være på rett plass
+ */
+const toggleBulletListOffBetweenListElements = (args: {
+  draft: Draft<LetterEditorState>;
+  itemContentIndex: ItemContentIndex;
+}) => {
+  const thisBlock = args.draft.redigertBrev.blocks[args.itemContentIndex.blockIndex];
+
+  const thisBlockContentBeforeItemList = thisBlock.content.slice(0, args.itemContentIndex.contentIndex);
```

Comment:
Vurdere å skrive denne om til splicing også? 
- En av de nye listene bør beholde original ID 
- Den som beholder ID må også beholde deletedItems, pluss det itemet som toggles må legges til.
- De ternære uttrykkene for `thisBlockContentBeforeItemList` og `thisBlockContentAfterItemList` burde ikke være nødvendig.

Reply:
yep, ble bare ikke helt ferdig :)

Reply:
Jeg lurer på om splicing her bare gjør det vanskeligere enn det trengs? Siden vi må uansett splitte arrayet i 2, for at punkten skal havne i mellom.

Reply:
Ja, sikkert greit det. Må beholde id på en av dem i hvertfall.

Reply:
Her er hvordan det kan se ut med slice
```
  const thisItemList = thisBlock.content[args.itemContentIndex.contentIndex] as ItemList;
  const thisItem = thisItemList.items[args.itemContentIndex.itemIndex];
  const itemsAfter = thisItemList.items.splice(args.itemContentIndex.itemIndex, thisItemList.items.length).slice(1);
  deleteElements(itemsAfter, thisItemList.items, thisItemList.deletedItems);
  thisBlock.content.splice(args.itemContentIndex.contentIndex + 1, 0, newItemList({ items: itemsAfter }));
  thisBlock.content.splice(args.itemContentIndex.contentIndex, 0, ...thisItem.content);

```